Refactoring Repo Loading and Moving/Renaming Git Files

Aryan-Barbarian opened the pull request:
In reference to #155 and #146 this pull request makes it much easier to load a git repo and start using it. There will never be any instances of the program thinking it has a repo locally (and can start processing it into a GitRepo object) when it actually doesn't.

As outlined in the README, it allows the user to load a repo by its name (e.g. "scipy"), by its path locally, or by its git url. It will not reload repos unnecessarily. 

However, this changes Repos.py to RepoLoader.py. This change will be inconvenient, so I can definitely change it back if need be.

sbenthall commented:
This is pending changes to the notebooks that depend on this code so they use the right module name. Otherwise, all the changes are great and good to go!

Aryan-Barbarian commented:
Some changes regarding the last two commits:

1. I moved all of the files in git_data to bigbang. Everything related to git is now part of the main bigbang library. #160 
2. I changed the git python files' naming convention from CamelCase (`RepoLoader.py`) to lowercase with underscores (`repo_loader.py`) to fit in with the other bigbang files. #171 
3. I edited all of the notebooks to work with this new naming convention. #179 
4. I moved the `sample_git_repos` folder `to archives/`. I believe this, combined with the previous changes, pretty much solves #146, #155, #158.

Aryan-Barbarian commented:
I realize that the travis build is failing, but I created a fresh conda install on my computer and installed everything from scratch and did not run into any problems.
